java/code/src/com/suse/manager/webui/controllers/virtualization/VirtualNetsController.java

```diff
@@ -98,11 +98,14 @@ public void initRoutes(JadeTemplateEngine jade) {
      */
     public ModelAndView show(Request request, Response response, User user) {
         Server host = getServer(request, user);
+        String minionId = host.asMinionServer().orElseThrow(NotFoundException::new).getMinionId();
```

Comment:
I don't really understand why you say there would be an NPE here... the `getServer()` function above throws a `NotFoundException` if the server doesn't exist, so we only need to care about the server being a Salt minion on that line.

Adding an exception only if the server is not a minion is an interesting idea. That would mean we also add a route to handle it in the global spark routes. However this URL is only exposed in the UI if the server is a minion... Does it make sense to add this only for the case the user tweaks some copied URL from another page? In the worst case he'll get the usual 404.

Reply:
if getServer throws an exception there wont be an NPE so that's fine. As for the case its not a minion if its a case that does not usually happen then having a confusing exception thrown may not be a big issue.